tests/integration/test_kfp_functional_v1.py

```diff
@@ -74,35 +106,39 @@ async def test_build_and_deploy(ops_test: OpsTest, request, lightkube_client):
         raise_on_blocked=False,  # These apps block while waiting for each other to deploy/relate
         raise_on_error=True,
         timeout=3600,
-        idle_period=30,
+        idle_period=120,
```

Comment:
Regarding this, I had changed this to `30` [here](https://github.com/canonical/kfp-operators/pull/350/files#diff-ad79bdbee1716da8d0eebee39133a7e79767b69755aef15512553ad4a1588ad8L77-R77) since this made the CI flaky and assertion was failing although the charms were all active. I remember having to run the tests many times (both in the CI and locally) in order for them to succeed.

Reply:
Just to clarify, they were failing because one of the charms would go to `Maintenance` status during this long `idle_period`, and the assertion would end up timing out.

Reply:
It was not failing for me, in fact the bundle tests ran and succeeded at the first time.